neqo-helper/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::{
+    io::{self, Error, ErrorKind, IoSlice, IoSliceMut},
+    net::{SocketAddr, UdpSocket},
+    process::exit,
+};
+
+use nix::{
+    cmsg_space,
+    errno::Errno::{EAGAIN, EINTR},
+    sys::socket::{
+        recvmsg, sendmsg, setsockopt,
+        sockopt::{
+            IpDontFrag, IpRecvTos, IpRecvTtl, IpTos, IpTtl, Ipv6DontFrag, Ipv6HopLimit,
+            Ipv6RecvHopLimit, Ipv6RecvTClass, Ipv6TClass,
+        },
+        AddressFamily, ControlMessage, ControlMessageOwned, MsgFlags, SockaddrLike,
+        SockaddrStorage,
+    },
+};
+
+use neqo_common::Datagram;
+
+// Bind a UDP socket and set some default socket options.
+pub fn bind(local_addr: SocketAddr) -> io::Result<UdpSocket> {
```

Comment:
The alternative to creating a new crate is to add this to a new module in neqo-common.  A non-default feature flag that makes this module conditionally available would ensure that mainline code doesn't need to pull in the nix and libc dependencies.

A new crate isn't that big of a deal if you only use it for the command-line tools, but I can see how we might want to use these facilities in Firefox proper, for which having it in neqo-common might help.

Reply:
Thanks for the feedback. Will need to read up on those concepts.

Reply:
@martinthomson I think I made that change.